### PR TITLE
Other user's profile

### PR DESCRIPTION
**Pull request by SaloniSS**
# Description

Alternate placeholders: stats contains your profile and challenges contains user profile (googleID as props needs to be passed)

### CODE REVIEW

File: app/screens/FriendProfile.tsx [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+import { AuthContext } from "../AuthContext";
+import ProfileInfo from "../components/ProfileInfo";
+import Post from "../components/PostList";
+
+const axios = require("axios").default;
+
+const FriendProfile = (props) => {
+  const auth = useContext(AuthContext);
+  let [user, setUser] = useState(null);
+  let [self, setSelf] = useState(null);
```
Comment (ReshmiCode) [inline app/screens/FriendProfile.tsx:14]: self is unused so not needed?

File: app/screens/FriendProfile.tsx [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+      {
+        Following: newFollowing,
+      }
+    );
+    console.log(`Following user ${user.googleID} now`);
+    await fetchData();
+  }, [refreshing]);
+
+  const unfollowUser = React.useCallback(async () => {
+    console.log(`Unfollowing user ${user.googleID} now`);
```
Comment (ReshmiCode) [inline app/screens/FriendProfile.tsx:62]: no logic here?

File: app/screens/FriendProfile.tsx (hunk 1 above)
Comment (ReshmiCode) [reply app/screens/FriendProfile.tsx:14]: ideally the profile for yourself and your friends would be the same component just what is rendered like the follow button is conditional based on this self var but if that's too much work it's fine

Comment (SaloniSS) [reply app/screens/FriendProfile.tsx:14]: lmao... I forgot to remove that

File: app/screens/FriendProfile.tsx (hunk 2 above)
Comment (SaloniSS) [reply app/screens/FriendProfile.tsx:62]: I forgot to push... Oops

File: app/screens/FriendProfile.tsx (hunk 1 above)
Comment (SaloniSS) [reply app/screens/FriendProfile.tsx]: I didn’t want to clutter that screen with that logic... Moreover your profile doesn’t require a prop and this takes a prop

File: app/screens/FriendProfile.tsx [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+  let [refreshing, setRefreshing] = useState(false);
+
+  async function fetchData() {
+    setRefreshing(true);
+    const result = await axios(
+      `https://productivitree.wl.r.appspot.com/api/v1/users/${props.user}`
+    );
+    setUser(result.data.payload);
+    const response = await axios(
+      `https://productivitree.wl.r.appspot.com/api/v1/users/${auth.googleID}`
```
Comment (ReshmiCode) [inline app/screens/FriendProfile.tsx:25]: why is both line 21 and 25 needed? isn't line 21 where we should be getting following/followers from

File: app/screens/FriendProfile.tsx [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+
+  useEffect(() => {
+    fetchData();
+  }, []);
+
+  const onRefresh = React.useCallback(async () => {
+    await fetchData();
+  }, [refreshing]);
+
+  const followUser = React.useCallback(async () => {
```
Comment (ReshmiCode) [inline app/screens/FriendProfile.tsx:40]: isn't useCallback only needed on a function if you call it inside of useEffect but define it outside

File: app/screens/FriendProfile.tsx (hunk 3 above)
Comment (ReshmiCode) [reply app/screens/FriendProfile.tsx:25]: oo it's to patch your own followers/following. We could add a backend endpoint so you wouldn't need that but ig it's too late

File: app/screens/FriendProfile.tsx (hunk 2 above)
Comment (ReshmiCode) [reply app/screens/FriendProfile.tsx:61]: uh I still don't see it

**Comment by SaloniSS** [comment]
image

File: app/screens/FriendProfile.tsx (hunk 3 above)
Comment (SaloniSS) [reply app/screens/FriendProfile.tsx:25]: Line 21 is to get the person’s screen, line 25 is to get your following/follower

File: app/screens/FriendProfile.tsx (hunk 4 above)
Comment (SaloniSS) [reply app/screens/FriendProfile.tsx:40]: I just used the template for inRefresh since the screen needs to be refreshed after those actions

**Pull request merged**